crates/driver/src/infra/mempool/mod.rs

```diff
@@ -10,6 +10,10 @@ pub async fn send(
     solver: &Solver,
     settlement: settlement::Simulated,
 ) -> Result<(), Error> {
+    if mempools.is_empty() {
+        // handle this case specificially because `select_ok` panics otherwise.
+        return Err(Error::AllMempoolsFailed);
+    }
```

Comment:
Unrelated to this PR but could remember in an issue. If this fails with no mempools then we should catch that during startup.